### ISSUE TITLE
RSTUDIO hangs foerever ! 

### ISSUE THREAD

**Issue by neonira** [rstudio/rstudio#2971]
### System details
See diagnostic file attached. Apparently it is about detectParentTermination() ... 
[diagnostics-report.txt](https://github.com/rstudio/rstudio/files/2088032/diagnostics-report.txt)

### Steps to reproduce the problem
None. It happens under my new linux 18:04  ubuntu machine running under gnome-wayland GUI with multi workspace activated. 

### Describe the problem in detail
Rstudio hangs and does not respond anymore. Unable to type in any character. This happens while RSTUDIO is open for few minutes and not focussed in for a given time. I mean, just using another workspace than the one where RSTUDIO has been launched, seems to bring issue on RSTUDIO. 

### Describe the behavior you expected
Being able to type in character in console or RStudio editor file at least ..

**Comment by jmcphers** [comment]
We haven't been able to reproduce this. :-/ Thanks for the diagnostics report! Can you provide any other details?

1. Did it happen prior to upgrading to Ubuntu 18?
2. Does it only happen on Wayland?
3. Does it stop happening if you turn off multi-workspace?

**Comment by neonira** [comment]
1. Did it happen prior to upgrading to Ubuntu 18? I don't know.
   2. Does it only happen on Wayland? I don't know.
   3. Does it stop happening if you turn off multi-workspace? I don't know.

What I know, is that turning off colorization of the RSTUDIO console helps
a lot, as it is no more crashing ...

Hope that's help


2018-06-11 14:42 GMT+02:00 Jonathan <notifications@github.com>:

**Comment by jmcphers** [comment]
Curious! Can you start `rstudio` under `gdb` and collect a backtrace when it crashes? (let us know if you need the steps for doing that)

**Comment by neonira** [comment]
Will try to do so. Know that RSTUDIO was not more accessible. I believe
that it is the window manager that is not able to exchange with RSTUDIO.
By the way, I confirm that switching off the option of console input
colorization as ended the issue (for the moment).

Some stats - I crashed several time rstudio per day last week end. Refer
already provided file if you want the real number. Today, I worked under
RSTUDIO a full day, without any issue.



2018-06-11 20:24 GMT+02:00 Jonathan <notifications@github.com>:
